File: scratch/api-first-call/first_anthropic_call.ts
import "dotenv/config";
import Anthropic from "@anthropic-ai/sdk";

async function main() {
  const apiKey = process.env.ANTHROPIC_API_KEY;

  if (!apiKey) {
    throw new Error("ANTHROPIC_API_KEY is missing. Add it to your .env file.");
  }

  const client = new Anthropic({
    apiKey,
  });

  const response = await client.messages.create({
    model: "claude-sonnet-4-6",
    max_tokens: 256,
    messages: [
      {
        role: "user",
        content: "What is a neural network in one sentence?",
      },
    ],
  });

  const firstBlock = response.content[0];

  if (firstBlock.type === "text") {
    console.log(firstBlock.text);
  } else {
    console.log(firstBlock);
  }
}

main().catch((error) => {
  console.error("API call failed:");
  console.error(error);
  process.exit(1);
});

File: scratch/api-first-call/first_anthropic_call.py
import os
from dotenv import load_dotenv
import anthropic

load_dotenv()

api_key = os.getenv("ANTHROPIC_API_KEY")

if not api_key:
    raise RuntimeError("ANTHROPIC_API_KEY is missing. Add it to your .env file.")

client = anthropic.Anthropic(api_key=api_key)

response = client.messages.create(
    model="claude-sonnet-4-6",
    max_tokens=256,
    messages=[
        {
            "role": "user",
            "content": "What is a neural network in one sentence?"
        }
    ],
)

print(response.content[0].text)
